doc/Yang Model Guidelines.md

```diff
@@ -0,0 +1,248 @@
+#   SONiC YANG MODEL GUIDELINES
+
+### Rev 1.0 ###
+
+### Revision ###
+
+ | Rev |     Date    |       Author       | Change Description                |
+ |:---:|:-----------:|:------------------:|-----------------------------------|
+ | 1.0 | 22 Aug 2019             |      Praveen Chaudhary      | Initial version                    |
+ | 1.1 | 29 Aug 2019             |      Praveen Chaudhary      | Addressed review comments.
+
+
+
+###  This document lists the guidelines, which will be used to write YANG Modules for SONiC. These YANG Modules will be primarily based on or represent the ABNF Schema of SONiC, and the syntax of YANG models must follow RFC 7950.
+
+### These YANG models will be used to verify the configuration for SONiC switches, so a library which supports validation of configuration on SONiC Switch must use these YANG Models. List of such Libraries are: 1.) Configuration Validation Library. (CVL)
```

Comment:
Can we also add that the same YANG file is used at NB too, with possibly additional metadata as required ?